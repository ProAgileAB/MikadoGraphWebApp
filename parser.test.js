
const { parse } = require('./parser.js')

describe("parser", () => {
    it("can parse a single Done", () => {
        const result = parse("Done: Goal")
        expect(result).toStrictEqual([{'done': "Goal"}])
    })
    it("can parse a single Goal", () => {
        const result = parse("Goal\n  This is the goal\n")
        expect(result).toStrictEqual([{'goal': "This is the goal"}])
    })
    it("can parse a needs node", () => {
        const result = parse("Goal needs AddCalls\n  Call getScore function in both places\n")
        expect(result).toMatchSnapshot()
    })
})
