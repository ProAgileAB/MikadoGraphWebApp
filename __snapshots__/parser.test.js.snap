// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser can parse a needs node 1`] = `
Array [
  Object {
    "needs": "AddCalls",
    "steps": Array [
      "Goal",
    ],
    "text": "Call getScore function in both places",
  },
]
`;

exports[`parser can parse a realistic example 1`] = `
Array [
  Object {
    "goal": "Reduce duplication of switch logic",
  },
  Object {
    "needs": "Calls",
    "steps": Array [
      "Goal",
    ],
    "text": "Call getpointName(int) replacing both switches",
  },
  Object {
    "needs": "AddFunction",
    "steps": Array [
      "Calls",
    ],
    "text": "Add getPointName(int) function based on bottom switch",
  },
  Object {
    "needs": "AddAll",
    "steps": Array [
      "AddFunction",
    ],
    "text": "Add \\"+All\\" after top switch",
  },
  Object {
    "needs": "Format",
    "steps": Array [
      "AddAll",
    ],
    "text": "Format file for readability",
  },
  Object {
    "needs": "DeuceReturn",
    "steps": Array [
      "AddAll",
    ],
    "text": "Return \\"Deuce if score >= before top switch",
  },
  Object {
    "needs": "Gitignore",
    "steps": Array [
      "DeuceReturn",
      "Format",
    ],
    "text": "Update .gitignore to reduce workspace clutter",
  },
  Object {
    "done": "AddAll",
  },
  Object {
    "done": "Format",
  },
]
`;
